conv/format.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 go-swagger maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conv
+
+import (
+	"strconv"
+	"unsafe"
+)
+
+// FormatInteger turns an integer type into a string.
+func FormatInteger[T Signed](value T) string {
+	return strconv.FormatInt(int64(value), 10)
+}
+
+// FormatUinteger turns an unsigned integer type into a string.
+func FormatUinteger[T Unsigned](value T) string {
+	return strconv.FormatUint(uint64(value), 10)
+}
+
+// FormatFloat turns a floating point numerical value into a string.
+func FormatFloat[T Float](value T) string {
+	return strconv.FormatFloat(float64(value), 'f', -1, int(unsafe.Sizeof(value))*8)
+}
```

Comment:
The need of using unsafe here should be explained.

Also, couldn't you use a type assertion on value?

something like


```suggestion
// FormatFloat turns a floating point numerical value into a string.
func FormatFloat[T Float](value T) string {
	bitsSize := 64
	if _, ok := any(value).(float32): ok {
		bitsSize = 32
	}
	return strconv.FormatFloat(float64(value), 'f', -1, bitsSize)
}
```

Reply:
Just noticing you are using unsafe in multiple methods in fact

Reply:
Yes. It is the fastest way to achieve that afaik.
"unsafe.SizeOf" is not really unsafe after all. 
Contrary to your suggestion, SizeOf is optimized at compile time (at least for T in primitive types).

Perhaps I should add a comment to "justify" the use of this package ...

Reply:
added this explanation, keeping the call to SizeOf. Type assertion for integers would make the code uglier IMHO and slower for sure (okay, we're talking about ns here...  :) ).

Reply:
I also found these since I reported it

- https://github.com/golang/go/issues/67110#issuecomment-2089462535
- https://github.com/golang/go/issues/29982